Sources/NostrSDK/RelayPool.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  RelayPool.swift
+//
+//
+//  Created by Bryan Montz on 12/31/23.
+//
+
+import Combine
+import Foundation
+import os.log
+
+/// An object that manages a set of relays.
+/// Send events and messages to all the relays in the pool.
+public final class RelayPool: ObservableObject, RelayDelegate, RelayOperating {
+    
+    /// The set of relays.
+    public private(set) var relays = Set<Relay>()
+    
+    /// A Publisher that publishes all events from all relays.
+    @Published public private(set) var events = PassthroughSubject<RelayEvent, Never>()
+    
+    /// A delegate to receive events and state changes.
+    public weak var delegate: RelayDelegate?
+    
+    private let logger = Logger(subsystem: "NostrSDK", category: "RelayPool")
+    private var cancellable: AnyCancellable?
+    
+    public init(relays: Set<Relay> = [], delegate: RelayDelegate? = nil) {
+        self.relays = relays
+        self.delegate = delegate
+        setUpRelays()
+    }
+    
+    public convenience init(relayURLs: Set<URL> = [], delegate: RelayDelegate? = nil) {
+        self.init(relays: Set(relayURLs.compactMap { try? Relay(url: $0) }),
+                  delegate: delegate)
+    }
+    
+    private func setUpRelays() {
+        relays.forEach { setUpRelay($0) }
+        updateEventsPublisher()
+    }
+    
+    private func setUpRelay(_ relay: Relay) {
+        relay.delegate = self
```

Comment:
Is the thought of assigning `relay.delegate` to `self` rather than `delegate` directly such that if we wanted to add additional behavior in `RelayPool`, we could?

Reply:
Hmm. Actually I didn't even think of doing it that way. But it's functionally equivalent to what I did and simpler. I'll do that, thanks.